### ISSUE TITLE
crash server

### ISSUE THREAD

**Issue by teracube** [Heisenburger69/BurgerSpawners#98]
BurgerSpawners_dev-91
https://poggit.pmmp.io/r/113620/BurgerSpawners_dev-91.phar

Pmmp version 3.18.0
mcpe_version : 1.16.210
https://jenkins.pmmp.io/job/PocketMine-MP/lastSuccessfulBuild/artifact/PocketMine-MP.phar

```
2021-03-19 [20:56:52] [Server thread/CRITICAL]: InvalidArgumentException: "Unknown entity numeric ID minecraft:ravager" (EXCEPTION) in "plugins/BurgerSpawners.phar/src/Heisenburger69/BurgerSpawners/Pocketmine/AddActorPacket" at line 120
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #0 pmsrc/src/pocketmine/network/mcpe/protocol/DataPacket(127): Heisenburger69\BurgerSpawners\Pocketmine\AddActorPacket->encodePayload()
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #1 pmsrc/src/pocketmine/network/mcpe/RakLibInterface(244): pocketmine\network\mcpe\protocol\DataPacket->encode()
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #2 pmsrc/src/pocketmine/Player(3347): pocketmine\network\mcpe\RakLibInterface->putPacket(object pocketmine\Player, object Heisenburger69\BurgerSpawners\Pocketmine\AddActorPacket, boolean , boolean )
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #3 pmsrc/src/pocketmine/Player(3364): pocketmine\Player->sendDataPacket(object Heisenburger69\BurgerSpawners\Pocketmine\AddActorPacket, boolean , boolean )
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #4 plugins/BurgerSpawners.phar/src/Heisenburger69/BurgerSpawners/Entities/Ravager(38): pocketmine\Player->dataPacket(object Heisenburger69\BurgerSpawners\Pocketmine\AddActorPacket)
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #5 pmsrc/src/pocketmine/entity/Entity(1961): Heisenburger69\BurgerSpawners\Entities\Ravager->sendSpawnPacket(object pocketmine\Player)
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #6 pmsrc/src/pocketmine/Player(1021): pocketmine\entity\Entity->spawnTo(object pocketmine\Player)
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #7 pmsrc/src/pocketmine/level/Level(2490): pocketmine\Player->sendChunk(integer 20, integer 4, object pocketmine\network\mcpe\protocol\BatchPacket)
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #8 pmsrc/src/pocketmine/level/Level(2545): pocketmine\level\Level->sendChunkFromCache(integer 20, integer 4)
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #9 pmsrc/src/pocketmine/level/format/io/ChunkRequestTask(81): pocketmine\level\Level->chunkRequestCallback(integer 20, integer 4, object pocketmine\network\mcpe\protocol\BatchPacket)
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #10 pmsrc/src/pocketmine/scheduler/AsyncPool(319): pocketmine\level\format\io\ChunkRequestTask->onCompletion(object pocketmine\Server)
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #11 pmsrc/src/pocketmine/Server(2393): pocketmine\scheduler\AsyncPool->collectTasks()
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #12 pmsrc/src/pocketmine/Server(2152): pocketmine\Server->tick()
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #13 pmsrc/src/pocketmine/Server(1992): pocketmine\Server->tickProcessor()
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #14 pmsrc/src/pocketmine/Server(1586): pocketmine\Server->start()
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #15 pmsrc/src/pocketmine/PocketMine(291): pocketmine\Server->__construct(object BaseClassLoader, object pocketmine\utils\MainLogger, string[16] /home/container/, string[24] /home/container/plugins/)
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #16 pmsrc/src/pocketmine/PocketMine(321): pocketmine\server()
2021-03-19 [20:56:52] [Server thread/CRITICAL]: #17 pmsrc(11): require(string[71] phar:///home/container/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
2021-03-19 [20:56:52] [Server thread/EMERGENCY]: An unrecoverable error has occurred and the server has crashed. Creating a crash dump
2021-03-19 [20:56:52] [Server thread/EMERGENCY]: Please upload the "/home/container/crashdumps/Fri_Mar_19-20.56.52-CET_2021.log" file to the Crash Archive and submit the link to the Bug Reporting page. Give as much info as you can.
2021-03-19 [20:56:52] [Server thread/EMERGENCY]: The crash dump has been automatically submitted to the Crash Archive. You can view it on https://crash.pmmp.io/view/4833974 or use the ID #4833974.
```

```
PocketMine-MP Crash Dump Fri Mar 19 20:53:58 CET 2021

Error: Unknown entity numeric ID minecraft:ravager
File: plugins/BurgerSpawners.phar/src/Heisenburger69/BurgerSpawners/Pocketmine/AddActorPacket
Line: 120
Type: InvalidArgumentException

THIS CRASH WAS CAUSED BY A PLUGIN
BAD PLUGIN: BurgerSpawners v0.1.4

Code:
[111]         121 => "minecraft:fox",
[112]         59 => "minecraft:ravager",
[113]     ];
[114] 
[115]     protected function encodePayload(): void
[116]     {
[117]         $this->putEntityUniqueId($this->entityUniqueId ?? $this->entityRuntimeId);
[118]         $this->putEntityRuntimeId($this->entityRuntimeId);
[119]         if (!isset(self::LEGACY_ID_MAP_BC[$this->type])) {
[120]             throw new \InvalidArgumentException("Unknown entity numeric ID $this->type");
[121]         }
[122]         $this->putString(self::LEGACY_ID_MAP_BC[$this->type]);
[123]         $this->putVector3($this->position);
[124]         $this->putVector3Nullable($this->motion);
[125]         ($this->buffer .= (\pack("g", $this->pitch)));
[126]         ($this->buffer .= (\pack("g", $this->yaw)));
[127]         ($this->buffer .= (\pack("g", $this->headYaw)));
[128] 
[129]         $this->putUnsignedVarInt(count($this->attributes));
[130]         foreach ($this->attributes as $attribute) {

Backtrace:
#0 pmsrc/src/pocketmine/network/mcpe/protocol/DataPacket(127): Heisenburger69\BurgerSpawners\Pocketmine\AddActorPacket->encodePayload()
#1 pmsrc/src/pocketmine/network/mcpe/RakLibInterface(244): pocketmine\network\mcpe\protocol\DataPacket->encode()
#2 pmsrc/src/pocketmine/Player(3347): pocketmine\network\mcpe\RakLibInterface->putPacket(object pocketmine\Player, object Heisenburger69\BurgerSpawners\Pocketmine\AddActorPacket, boolean , boolean )
#3 pmsrc/src/pocketmine/Player(3364): pocketmine\Player->sendDataPacket(object Heisenburger69\BurgerSpawners\Pocketmine\AddActorPacket, boolean , boolean )
#4 plugins/BurgerSpawners.phar/src/Heisenburger69/BurgerSpawners/Entities/Ravager(38): pocketmine\Player->dataPacket(object Heisenburger69\BurgerSpawners\Pocketmine\AddActorPacket)
#5 pmsrc/src/pocketmine/entity/Entity(1961): Heisenburger69\BurgerSpawners\Entities\Ravager->sendSpawnPacket(object pocketmine\Player)
#6 pmsrc/src/pocketmine/Player(1021): pocketmine\entity\Entity->spawnTo(object pocketmine\Player)
#7 pmsrc/src/pocketmine/level/Level(2490): pocketmine\Player->sendChunk(integer 20, integer 4, object pocketmine\network\mcpe\protocol\BatchPacket)
#8 pmsrc/src/pocketmine/level/Level(2545): pocketmine\level\Level->sendChunkFromCache(integer 20, integer 4)
#9 pmsrc/src/pocketmine/level/format/io/ChunkRequestTask(81): pocketmine\level\Level->chunkRequestCallback(integer 20, integer 4, object pocketmine\network\mcpe\protocol\BatchPacket)
#10 pmsrc/src/pocketmine/scheduler/AsyncPool(319): pocketmine\level\format\io\ChunkRequestTask->onCompletion(object pocketmine\Server)
#11 pmsrc/src/pocketmine/Server(2393): pocketmine\scheduler\AsyncPool->collectTasks()
#12 pmsrc/src/pocketmine/Server(2152): pocketmine\Server->tick()
#13 pmsrc/src/pocketmine/Server(1992): pocketmine\Server->tickProcessor()
#14 pmsrc/src/pocketmine/Server(1586): pocketmine\Server->start()
#15 pmsrc/src/pocketmine/PocketMine(291): pocketmine\Server->__construct(object BaseClassLoader, object pocketmine\utils\MainLogger, string[16] /home/container/, string[24] /home/container/plugins/)
#16 pmsrc/src/pocketmine/PocketMine(321): pocketmine\server()
#17 pmsrc(11): require(string[71] phar:///home/container/PocketMine-MP.phar/src/pocketmine/PocketMine.php)

PocketMine-MP version: 3.18.0 [Protocol 428]
Git commit: 3f6490626319152dca07a5c61a1e94f8d5793d56
uname -a: Linux 643340b8-28d5-4090-bb05-7e3ac9a12b14 5.9.0-050900-generic #202010112230 SMP Sun Oct 11 22:34:01 UTC 2020 x86_64
PHP Version: 7.3.16
Zend version: 3.3.16
OS : Linux, linux
```

**Comment by iZeaoGamer** [comment]
You are on an incorrect build. The latest build should fix this issue.

**Comment by Heisenburger69** [comment]
Fixed in dev build #95
